src/decide.ts

```diff
@@ -102,6 +104,30 @@ export function compareFileTreesIgnoringNames(
 	);
 }
 
+export function compareFileTreesForPartialMatch(
+	candidate: Metafile,
+	searchee: Searchee,
+	minPartialMatch: number
+): boolean {
+	let matchedSizes = 0;
+	let availableFiles = searchee.files.slice();
+	candidate.files.forEach((elOfA) => {
+		let matchedFiles = availableFiles.filter((elOfB) => {
+			elOfA.length === elOfB.length;
+		});
+		if (matchedFiles.length > 1) {
+			matchedFiles = matchedFiles.filter((elOfB) => {
+				elOfA.name === elOfB.name;
+			});
+		}
+		if (matchedFiles.length) {
+			matchedSizes += elOfA.length;
+			availableFiles.splice(availableFiles.indexOf(matchedFiles[0]), 1);
+		}
+	});
```

Comment:
nit: `for (const elOfA of candidate.files) {}` is preferred over `Array.prototype.forEach`

Can you explain how this `compareFileTreesForPartialMatch` works at a high level?

Reply:
It takes each file in the candidate and looks for an exact size match in the searchee. For each file that matches, it's removed from the searchee pool and its size is added to a count. If the ratio of machedSizes to total candidate size is above minPartialMatch, it's successful.

If there are duplicate sizes (https://github.com/cross-seed/cross-seed/issues/610#issuecomment-1987101404), we try to get an exact name match to know which file is correct. If there are still multiple files, we just choose any as if both the size and name matches, it's probably safe. If the name doesn't match when there are duplicates, the file is not counted.

> so is partial effectively a superset of risky ?

Yes. They are the same if minPartialMatch is 1 or it's a single file torrent.

Reply:
I think availableFiles is redundant since due to the name check so it's now just going through searchee.files each time.

Reply:
availableFiles is redundant due to checking the file name so it's now using searchee.files each time.